homeassistant/components/tuya/climate.py

```diff
@@ -437,7 +438,10 @@ def hvac_mode(self) -> str:
             return HVAC_MODE_OFF
 
         if self.device.status.get(DPCode.MODE) is not None:
+            if self.device.status[DPCode.MODE] not in TUYA_HVAC_TO_HA.keys():
+                return HVAC_MODE_HEAT_COOL
```

Comment:
We should not default to HEAT_COOL, instead, we should show off.

I'm OK with adding additional modes to the lookup.

Reply:
we shouldn't show `off` when the device is operating

Reply:
We don't know if it is operating; the value is unknown.

Reply:
This is not `NULL`, but one of the unsupported values, like `eco`, `boost`, or `comfort`

Reply:
Seem my answer above.

It is an unknown value; we can't assume an operating mode at all. Simply because the operating mode is unknown. As this property is one of the few properties in Home Assistant that cannot be `NoneType` (and thus resulting in "unknown" state), `off` is closest to it.

I've opened up #62984 to handle that case and log it properly; and marked it for 2021.12.7.